### PR TITLE
#4496 gallery card changes

### LINKED ISSUE #4496: New NFT card
Although not a major update, it will improve the user interface perspective in upcoming explore and collection detail releases.
image

- new figma file: explore v0.9, new components - https://www.figma.com/file/OwAEyE0QDPn3tHXZbyot1o/Explore-v0.9?node-id=0%3A1&t=Mp5WHz4Ig8kbcIsy-1
- dynamic scalling, i think currently we have pretty good working version on the landing page. 
- screenshot is only the minimum size card. Its meant for #4444 - keeping the gutter fixed and based on the minimum card size changing the amount of the cards. 

- the card will scale up same as the current one. Keeping the font sizes (16px and 12px) 
- fixed padding, also the same. 

### hover effects
- on nft name - changing from k-black to k-hovergrey, in darkmode from k-white to k-shade, there should then popup nft card #4340 
-  on "collection"- changing from grey to black, in dark mode to k-white (from the same color)

basically we are only changing the text inside and its color.

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4496
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-12-15 at 8 59 03 PM
Screen Shot 2022-12-15 at 8 59 13 PM

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Oh no.. CarouselCard, I find it misleading.
Can't we have a fresh `GalleryCard` component?

**Comment by prachi00** [comment]
> Oh no.. CarouselCard, I find it misleading.
> 
> Can't we have a fresh `GalleryCard` component?

Or can we rename carousel card and use it in both places? Since the ui and functionality is totally same

**Comment by roiLeo** [comment]
> Or can we rename carousel card and use it in both places? Since the ui and functionality is totally same

if the design is exactly the same I'm not against, as long as we don't end up with special cases, I'm fine with it.
Or the best would be to to create GalleryCard component in histoire that will be used by carousel & explore layout.

**Comment by exezbcz** [comment]
image
price should not be bold
- the hover on the nft name is for the hover card which isnt implemented yet #4340

**Comment by prachi00** [comment]
@roiLeo so I shifted the gallery card to histoire but I haven't used that yet for carousel, I feel like shifting carousel should be done in a new issue as there is some logical part for that, that might need some moving too 
so currently, carousel stays the same like before and the gallery one uses the new histoire component

**Review by roiLeo — CHANGES_REQUESTED** [review]
Hey! Why did you change the carousel code if the outcome is not affected?

### what needs to change in histoire component:
- use placeholder/fixed data
- don't use nuxt 
- don't use buefy helpers
- rename `NftCard` to `NeoCard`?

I don't see the point of `components/rmrk/Gallery/NftCard.vue` file you can directly import the component

**Comment by yangwao** [comment]
@prachi00  Can we please move it outside of `rmrk` folder? 🥹

https://github.com/kodadot/nft-gallery/pull/4511/files#diff-bc15d2a15f1220c1ae5eaa62bb0d747486a1cee56b21204b5439c46d7dd2e0ec

**Comment by prachi00** [comment]
> 
 I just did the UI changes for the carousel card that should be according to the design

**Comment by prachi00** [comment]
@roiLeo by buefy helpers, do you mean buefy css classes? is-flex and all?

**Review by roiLeo — COMMENTED** [review]
> @roiLeo by buefy helpers, do you mean buefy css classes? is-flex and all?

Nvm since we use Bulma theme in histoire it's working.

> Also, using histoire newNftCard in a component that is not in vue compostion (gallery.vue) doesnt work, that is why I created a new composition component and used it there

lol, Nice move! I didn't knew that Gallery wasn't rewritten in Vue3 way.

- What's the difference between `NeoNftCard` & `NftCard`? if one of them is not used please delete it
- Could you please create story for each component you create?
- `components/rmrk/Gallery/NftCard.vue` stil inside `rmrk` folder

**Comment by prachi00** [comment]
> > @roiLeo by buefy helpers, do you mean buefy css classes? is-flex and all?
> 
> Nvm since we use Bulma theme in histoire it's working.
> 
> > Also, using histoire newNftCard in a component that is not in vue compostion (gallery.vue) doesnt work, that is why I created a new composition component and used it there
> 
> lol, Nice move! I didn't knew that Gallery wasn't rewritten in Vue3 way.
> 
> * What's the difference between `NeoNftCard` & `NftCard`? if one of them is not used please delete it
> * Could you please create story for each component you create?
> * `components/rmrk/Gallery/NftCard.vue` stil inside `rmrk` folder

so NftCard is in vue composition, so it is using NeoNftCard which gallery is not able to yet as it isnt in vue3, I guess we can remove NftCard once gallery is in vue3 and then directly use NeoNftCard in gallery

**Comment by roiLeo** [comment]
> so NftCard is in vue composition...

I'm referring to `NeoNftCard` & `NftCard` component in histoire

**Comment by prachi00** [comment]
@roiLeo I added the story file, but I dont know how to see it, I ran story:dev 
But I dont see any of the components, its just loading forever
Screen Shot 2022-12-21 at 10 18 37 PM

File: libs/ui/src/components/NeoNftCard/NeoNftCard.story.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <Story title="NeoNftCard">
+    <Variant title="Nft Card">
+      <NeoNftCard :nft="nft" prefix="'rmrk'" :show-price="true" />
```
Comment (roiLeo) [inline libs/ui/src/components/NeoNftCard/NeoNftCard.story.vue:4]:
```suggestion
      <NeoNftCard :nft="nft" prefix="rmrk" show-price />
```

File: components/shared/gallery/NftCard.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+</template>
+
+<script lang="ts" setup>
+import { NeoNftCard } from '@kodadot1/brick'
+import type { NFT } from '@/components/rmrk/service/scheme'
+
+const { urlPrefix } = usePrefix()
+
+const props = defineProps<{
+  nft?: NFT
```
Comment (roiLeo) [inline components/shared/gallery/NftCard.vue:12]: optional prop?
```suggestion
  nft: NFT
```

File: libs/ui/src/components/NeoNftCard/NeoNftCard.vue [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  </div>
+</template>
+
+<script lang="ts" setup>
+import MediaItem from '../MediaItem/MediaItem.vue'
+import CommonTokenMoney from '@/components/shared/CommonTokenMoney.vue'
+import type { NFT } from '@/components/rmrk/service/scheme'
+
+defineProps<{
+  nft?: NFT
```
Comment (roiLeo) [inline libs/ui/src/components/NeoNftCard/NeoNftCard.vue:41]: same thing here
```suggestion
  nft: NFT
```

Comment (roiLeo) [reply libs/ui/src/components/NeoNftCard/NeoNftCard.vue:41]: ref #4544, missing `image`; `animationUrl`; `mimeType`; `collectionName`

**Review by roiLeo — APPROVED** [review]
✅ let's move on

**Comment by roiLeo** [comment]
wfm too, might need to wrap Story in grid layout
Screenshot 2022-12-22 at 09-02-22 NeoNftCard › Nft Card Histoire

**Comment by yangwao** [comment]
pay 80 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 24.65 USD/KSM ~ 3.245 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x9ab254460bf51eabb19888b2e20f862217adcb795c02dcbf80e3f73ffd42ff58](https://kusama.subscan.io/extrinsic/0x9ab254460bf51eabb19888b2e20f862217adcb795c02dcbf80e3f73ffd42ff58)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**